File: app/src/main/java/com/morozione/clicker/RecordManager.java
package com.morozione.clicker;

import android.graphics.Color;
import android.widget.TextView;

/**
 * Created by morozione on 07/19/17.
 */

public class RecordManager {

    public static void textForRecordDialog(int record, int newResult, TextView tvMessage, TextView tvResult) {
        if (newResult > record) {
            tvMessage.setText("New Record!!!");
            tvMessage.setTextColor(Color.parseColor("#FF3AA63A"));
            tvResult.setTextColor(Color.parseColor("#FF3AA63A"));
        } else {
            if (newResult > (record / 4 * 3)) {
                tvMessage.setText("Not Bad");
                tvMessage.setTextColor(Color.parseColor("#FF989608"));
                tvResult.setTextColor(Color.parseColor("#FF989608"));
            } else if (newResult > (record / 4 * 2)) {
                tvMessage.setText("So-So");
                tvMessage.setTextColor(Color.parseColor("#FF989608"));
                tvResult.setTextColor(Color.parseColor("#FF989608"));
            } else if (newResult > (record / 4)) {
                tvMessage.setText("Bad");
                tvMessage.setTextColor(Color.parseColor("#FFB01111"));
                tvResult.setTextColor(Color.parseColor("#FFB01111"));
            } else {
                tvMessage.setText("You Loser!!");
                tvMessage.setTextColor(Color.parseColor("#FFB01111"));
                tvResult.setTextColor(Color.parseColor("#FFB01111"));
            }
        }
        tvResult.setText(newResult + "");
    }

}

File: app/src/main/java/com/morozione/clicker/RecordManager.kt
package com.morozione.clicker

import androidx.compose.ui.graphics.Color

data class RecordFeedback(
    val message: String,
    val color: Color
)

object RecordManager {
    fun getRecordFeedback(record: Int, newResult: Int): RecordFeedback {
        return when {
            newResult > record -> RecordFeedback(
                message = "New Record!!!",
                color = Color(0xFF3AA63A)
            )
            newResult > (record * 3 / 4) -> RecordFeedback(
                message = "Not Bad",
                color = Color(0xFF989608)
            )
            newResult > (record / 2) -> RecordFeedback(
                message = "So-So",
                color = Color(0xFF989608)
            )
            newResult > (record / 4) -> RecordFeedback(
                message = "Bad",
                color = Color(0xFFB01111)
            )
            else -> RecordFeedback(
                message = "You Loser!!",
                color = Color(0xFFB01111)
            )
        }
    }
} 